### PR TITLE
TST: add 2d instruments to test_instrument

### PR DESCRIPTION
**Pull request by jklenzing**
This is a start at fixing the test coverage.  Still a couple of tests that don't pass for the 2d xarray objects.

### CODE REVIEW

**Comment by jklenzing** [comment]
The two tests that aren't working are the datetime slice tests, since pysat_testing2d uses a 100s cadence by default, and the slice test assumes a 1s cadence.  This could be fixed by implementing #389, htough probably outside of the scope here.  I don't think these two tests affect anything in this pull.

**Comment by jklenzing** [comment]
Two errors in the higher order tests.  These didn't appear before since those lines weren't being run.

File: pysat/tests/test_instrument.py [hunk 1]
```diff
@@ -345,7 +346,7 @@ def test_basic_repr(self):
         # No loaded data
         assert output.find('No loaded data') > 0
         assert output.find('Number of variables:') < 0
-        assert output.find('dummy') < 0
+        assert output.find('uts') < 0
```
Comment (jklenzing) [inline pysat/tests/test_instrument.py:349]: 'dummy' not present in 2d data.  switching to uts for the variable check as this is present in all instruments.

File: pysat/tests/test_instrument.py [hunk 2]
```diff
@@ -670,9 +675,9 @@ def test_basic_variable_renaming_lowercase(self, values):
                                          'alt_profiles':
                                              {'density': 'volume'}}])
     def test_ho_pandas_variable_renaming(self, values):
-        # check for pysat_testing2D instrument
+        # check for pysat_testing2d instrument
         if self.testInst.platform == 'pysat':
-            if self.testInst.name == 'testing2D':
+            if self.testInst.name == 'testing2d':
```
Comment (rstoneback) [inline pysat/tests/test_instrument.py:680]: ahhh. thanks!

File: pysat/tests/test_instrument.py [hunk 3]
```diff
@@ -64,7 +64,8 @@ def test_basic_instrument_load_no_input(self):
             self.testInst.load()
 
     def test_basic_instrument_load_by_file_and_multifile(self):
-        testInst = pysat.Instrument(platform='pysat', name='testing',
+        testInst = pysat.Instrument(platform=self.testInst.platform,
```
Comment (rstoneback) [inline pysat/tests/test_instrument.py:67]: Thanks.

File: pysat/tests/test_instrument.py [hunk 4]
```diff
@@ -477,6 +478,8 @@ def test_data_access_by_datetime_and_name(self):
                       self.testInst.data['uts'].values[0])
 
     def test_data_access_by_datetime_slicing_and_name(self):
+        if self.testInst.name == 'testing2d':
+            pytest.skip('not supported for 2d')
```
Comment (rstoneback) [inline pysat/tests/test_instrument.py:482]: perhaps 'not supported for 2d' due to limited time sampling?

Comment (rstoneback) [reply pysat/tests/test_instrument.py:482]: could try something like...
```suggestion
            time_step = (self.testInst.index[1] - self.testInst.index[0]).value/1.E9
            offset = pds.DateOffset(seconds=10*time_step)
            start = dt.datetime(2009, 1, 1, 0, 0, 0)
            stop = start + offset
```

File: pysat/tests/test_instrument.py [hunk 5]
```diff
@@ -583,6 +586,8 @@ def test_setting_partial_data_by_datetime_and_name(self):
         assert np.all(self.testInst[0, 'doubleMLT'] == 0)
 
     def test_setting_partial_data_by_datetime_slicing_and_name(self):
+        if self.testInst.name == 'testing2d':
+            pytest.skip('not supported for 2d')
```
Comment (rstoneback) [inline pysat/tests/test_instrument.py:590]: same as one above

File: pysat/tests/test_instrument.py [hunk 6]
```diff
@@ -1035,6 +1040,25 @@ def teardown(self):
         del self.testInst
 
 
+# ------------------------------------------------------------------------------
+#
+# Repeat tests above with 2d data
+#
+# ------------------------------------------------------------------------------
+class TestBasics2D(TestBasics):
```
Comment (rstoneback) [inline pysat/tests/test_instrument.py:1048]: Thank you for noticing that whole class was missed here.

File: pysat/tests/test_instrument.py [hunk 7]
```diff
@@ -583,6 +586,8 @@ def test_setting_partial_data_by_datetime_and_name(self):
         assert np.all(self.testInst[0, 'doubleMLT'] == 0)
 
     def test_setting_partial_data_by_datetime_slicing_and_name(self):
+        if self.testInst.name == 'testing2d':
+            pytest.skip('not supported for 2d')
         self.testInst.load(2009, 1)
         self.testInst['doubleMLT'] = 2. * self.testInst['mlt']
```
Comment (rstoneback) [inline pysat/tests/test_instrument.py:592]:
```suggestion
        self.testInst['doubleMLT'] = 2. * self.testInst['mlt']
        time_step = (self.testInst.index[1] - self.testInst.index[0]).value/1.E9
        offset = pds.DateOffset(seconds=10*time_step)
        start = dt.datetime(2009, 1, 1, 0, 0, 0)
        stop = start + offset
```

File: pysat/tests/test_instrument.py [hunk 8]
```diff
@@ -583,6 +586,8 @@ def test_setting_partial_data_by_datetime_and_name(self):
...
         self.testInst[dt.datetime(2009, 1, 1, 0, 0, 0):
```
Comment (rstoneback) [inline pysat/tests/test_instrument.py:593]:
```suggestion
        self.testInst[start:stop, 'doubleMLT'] = 0
```

**Comment by jklenzing** [comment]
Fixed.  This just leaves the assertion errors
```
FAILED pysat/tests/test_instrument.py::TestBasics2D::test_ho_pandas_variable_renaming_lowercase[values0] - AssertionError: assert 'ionization' in                      Ionization dummy_str ...
FAILED pysat/tests/test_instrument.py::TestBasics2D::test_ho_pandas_variable_renaming_lowercase[values1] - AssertionError: assert 'masa' in                           MASa dummy_str dummy_u...
```

**Comment by jklenzing** [comment]
I think all that's left here are the assertion errors in the new tests.  Not sure how to handle these.

**Comment by rstoneback** [comment]
Thanks Jeff! Those errors are on my list just not quite there yet :)

**Comment by rstoneback** [comment]
Thanks again Jeff. I added a commit to fix my test. I think we are good to go here. I'll take another pass for review.

**Comment by jklenzing** [comment]
coverage issues fixed.  merging.

**Pull request merged**